### PR TITLE
Portfolio CPT: Ensure Jetpack Settings page UI is clear for WoA sites with portfolio theme support

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes https://github.com/Automattic/vulcan/issues/523

## Proposed changes:

* The issue was that on WoA sites there is a new behaviour - introduced intentionally here: https://github.com/Automattic/jetpack/pull/39431 - whereby if using a theme that supports Portfolios CPTs, it isn't possible to toggle Portfolios off.
* This PR adds a message to the toggle UI to explain this and makes the toggle untoggleable (on the Jetpack > Settings > Writing page)
* One option would also have been to create an option to store the information so it was available via the React-based admin page, but to try and minimize performance issues I opted for the conditional creation of a variable.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/523

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Follow the testing instructions here: https://github.com/Automattic/jetpack/pull/39268

On a WoA site:
* When activating a theme that supports the Portfolio CPT (Sketch, Illustrator, Blask), the toggle at Jetpack > Settings > Writing (`/wp-admin/admin.php?page=jetpack#/writing`) should be disabled in the on position. You should also see some additional text in the description, explaining that the theme supports Jetpack Portfolios.
* This change will only show if the PR is applied with the Jetpack plugin - if only applied on the WordPress.com Features plugin then behaviour would be the same as with trunk.

Testing on self-hosted, there should be no changes in comparison to testing with trunk.

Testing on Simple, there should be no changes in comparison to testing with trunk.

On WoA with theme support:
Screenshot 2024-09-24 at 14 44 08

### CODE REVIEW

File: projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php [hunk 1]
```diff
@@ -67,6 +67,19 @@ public function __construct() {
 			} else {
 				add_action( 'init', array( $this, 'maybe_register_cpt' ) );
 			}
+
+			// Add a variable with the theme support status for the Jetpack Settings Portfolio toggle UI.
+			if ( current_theme_supports( self::CUSTOM_POST_TYPE ) ) {
```
Comment (fgiannar) [inline projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php:72]: Would it make sense to only register this script for WoA sites?

**Review by fgiannar — APPROVED** [review]
Thanks for working on this, Karen 👍 

Works as described on WoA sites where the admin interface style is set to `Classic`, aka managed via WP Admin.

Left an inline comment but not a blocker.

Noting that I can still toggle the Portfolio setting in Calypso Blue, aka on WoA sites  where the admin interface style is set to `Default` but I believe this out of the context of the current PR:

Screenshot 2024-09-25 at 11 56 15

File: projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php (hunk 1 above)
Comment (coder-karen) [reply projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php:72]: I thought so too initially. My concern was around the possibility of a WoA site moving to self hosted and somehow having cached values here. It is probably quite an edge case though. Another alternative would be to possibly use `wp_deregister_script` if a site is not a WoA site.
I think the current solution is a reasonable middle ground given that themes with support is likely (again a guess though) a smaller group than WoA sites - and if a self hosted site didn't have theme support for portfolios then for them to somehow have cached values they would moved from WoA and switched themes to move away from Portfolios 😅 . I'm possibly over-thinking this though. What do you think?

Comment (fgiannar) [reply projects/packages/classic-theme-helper/src/custom-post-types/class-jetpack-portfolio.php:72]: Thanks for the explanation. As you mentioned, themes with support is likely a smaller group than WoA sites, so I think it's ok to proceed with the current approach

**Comment by coder-karen** [comment]
@fgiannar thanks for the review! 

> Noting that I can still toggle the Portfolio setting in Calypso Blue, aka on WoA sites where the admin interface style is set to Default but I believe this out of the context of the current PR:

Yes I do think that would be confusing, so plan to follow up with it, noted it here: https://github.com/Automattic/vulcan/issues/523#issuecomment-2371338281

**Pull request merged**